Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -869,18 +870,25 @@ std::vector<std::shared_ptr<FitPeaksAlgorithm::PeakFitResult>> FitPeaks::fitPeak
       std::shared_ptr<FitPeaksAlgorithm::PeakFitResult> fit_result =
           std::make_shared<FitPeaksAlgorithm::PeakFitResult>(m_numPeaksToFit, numfuncparams);
 
-      fitSpectrumPeaks(static_cast<size_t>(wi), expected_peak_centers, fit_result, lastGoodPeakParameters);
+      size_t spectrum_peaks_not_enough_datapoints{0};
+      fitSpectrumPeaks(static_cast<size_t>(wi), expected_peak_centers, fit_result, lastGoodPeakParameters,
+                       spectrum_peaks_not_enough_datapoints);
 
       PARALLEL_CRITICAL(FindPeaks_WriteOutput) {
         writeFitResult(static_cast<size_t>(wi), expected_peak_centers, fit_result);
         fit_result_vector[wi - m_startWorkspaceIndex] = fit_result;
+        all_spectra_peaks_not_enough_datapoints += spectrum_peaks_not_enough_datapoints;
       }
       prog.report();
     }
     PARALLEL_END_INTERRUPT_REGION
   }
   PARALLEL_CHECK_INTERRUPT_REGION
 
+  if (all_spectra_peaks_not_enough_datapoints > 0)
+    g_log.notice() << all_spectra_peaks_not_enough_datapoints << " peaks rejected: not enough X(Y) datapoints."
+                   << std::endl;
```

Comment:
This is a minor performance thing. Using `std::endl` forces the logging system to flush buffers which incurs a performance hit. Changing it to us `"\n"` instead lets logging sort itself out. It is a bigger deal in other things, but to be consistent this should be
```suggestion
    g_log.notice() << all_spectra_peaks_not_enough_datapoints << " peaks rejected: not enough X(Y) datapoints.\n";
```